src/main/java/Duke.java

```diff
@@ -1,10 +1,122 @@
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(logo);
+
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+
+        Duke.runDuke();
+    }
+
+    // Carry out Add, List, Done commands if entered by user
+    // Terminates when user gives exit signal
+    private static void runDuke() {
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            String nextInput = scanner.nextLine();
+            // Validation check
+            if (nextInput.equals("bye")) {
+                Duke.stopDuke();
+                break;
+            }
+            // Handle List command if any
+            if (nextInput.equals("list")) {
+                // fflush
+                //System.out.println("");
+                Duke.handleCommandList();
+                continue;
+            }
+
+            // Check and Handle Done command
+            // Possible source of error
+            String[] commandArgs = nextInput.split(" ");
+
+            if (commandArgs[0].equals("done")) {
+                // Todo: Add exception handling
+                handleCommandDone(Integer.parseInt(commandArgs[1]));
+                continue;
+            }
+
+            if (commandArgs[0].equals("delete")) {
+                // Todo: Add exception handling
+                handleCommandDelete(Integer.parseInt(commandArgs[1]));
+                continue;
+            }
+
+            // Handle Add Command
+            Task t = null;
+            try {
+                switch (commandArgs[0]) {
+                    case "event":
+                        t = Event.createTask(commandArgs);
+                        break;
+                    case "todo":
+                        t = Todo.createTask(commandArgs);
+                        break;
+                    case "deadline":
+                        t = Deadline.createTask(commandArgs);
+                        break;
+                    default:
+                        // Unrecognised command
+                        throw new IllegalCommandException();
+                }
+            } catch (IllegalArgumentException e1){
+                System.out.println("    " + e1.getMessage());
+                continue;
+            } catch (IllegalCommandException e2) {
+                System.out.println("    " + e2.getMessage());
+                continue;
+            }
+
+            Duke.handleCommandAdd(t);
+        }
+    }
+
+    // Print a closing message before stopping Duke
+    private static void stopDuke() {
+        String closingMessage = "Bye. Hope to see you again soon!";
+        System.out.println("    " + closingMessage);
+    }
+
+    // Todo: abstract away the following logic from main via new interface for command handlers
```

Comment:
Even though its only a comment, this line is pretty long. Maybe break it up into 2 lines :)